internal/pkg/template/workload.go

```diff
@@ -483,6 +489,114 @@ func (cfg *ALBListener) RulePaths() []string {
 	return rulePaths
 }
 
+// ConditionGroups returns a slice of all the Splits on the Conditions across multiple listener rules.
+func (cfg *ALBListener) ConditionGroups(wkldType string, isHTTPS bool) []string {
+	var groupCount []string
+	for _, rule := range cfg.Rules {
+		groupCount = append(groupCount, strconv.Itoa(len(rule.GenerateConditionGroups(wkldType, isHTTPS))))
+	}
+	return groupCount
+}
+
+// conditionGroup represents groups of conditions per listener rule.
+type conditionGroup struct {
+	AllowedSourceIps []string
+	Aliases          []string
+}
+
+// GenerateConditionGroups will split allowed source ips and aliases into separate condition groups per listener rule.
+// This is needed because Listener rule has quota of 5 conditions per rule.
+// https://docs.aws.amazon.com/elasticloadbalancing/latest/application/load-balancer-limits.html
+func (lr ALBListenerRule) GenerateConditionGroups(wkldType string, isHTTPS bool) []conditionGroup {
+	var groups []conditionGroup
+	remaining := lr.calculateRemainingConditions(wkldType, isHTTPS)
+	if len(lr.Aliases)+len(lr.AllowedSourceIps) <= remaining {
+		return append(groups, conditionGroup{
+			AllowedSourceIps: lr.AllowedSourceIps,
+			Aliases:          lr.Aliases,
+		})
+	}
+	if isHTTPS && len(lr.AllowedSourceIps) != 0 {
+		return lr.generateConditionsWithSourceIPsAndAlias(remaining - 1)
+	}
+	if isHTTPS {
+		return lr.generateConditionsWithAliasOnly(remaining)
+	}
+	return lr.generateConditionWithSourceIPsOnly(remaining)
+}
+
+func (lr ALBListenerRule) calculateRemainingConditions(wkldType string, isHTTPS bool) int {
+	rcPerRule := totalConditionsPerRule
+	if isHTTPS {
+		return calculateConditionsHTTPS(lr.Path, rcPerRule, len(lr.Aliases))
+	}
+	return calculateConditionsHTTP(lr.Path, rcPerRule, wkldType)
+}
+
+func calculateConditionsHTTPS(path string, rcPerRule int, aliasCount int) int {
+	if aliasCount == 0 {
+		rcPerRule = rcPerRule - 1
+	}
+	if path != rootPath {
+		return rcPerRule - 2
+	}
+	return rcPerRule - 1
+}
+
+func calculateConditionsHTTP(path string, rcPerRule int, wkldType string) int {
+	if wkldType == backendWkld {
+		rcPerRule = rcPerRule - 2
+	}
+	if path != rootPath {
+		return rcPerRule - 2
+	}
+	return rcPerRule - 1
+}
```

Comment:
It doesn't make a lot of sense to me for these "1" and "2" without looking at the CFN template. I think this requires a lot of assumptions for how we write our CFN templates which is fragile. But I also fully understand the reason why we want to do this: to fulfill the requirement that we fully utilize the "5" max condition values per listener rule, so that we will be less likely to exceed the at most "100" listener rules per LB.

This makes me rethink if we should sacrifice this to make the code robust and less likely to break for better maintainability 💭  In reality it is less likely they'll need that many aliases for __non-prod__ environment. I'm wondering if we could just assume the condition group num to be always "1" so that to greatly simplify the code. And we'll have "1" alias per listener rule. And the total number of listener rules we'll have is
```
totalNum = 0
for rule := range ${listener rules they define in the manifest)}
    totalNum += len(rule.allowed_ip} * len(rule.aliases)
```
That way though we create more listener rules, we can simplify this `template/workload.go` file as well as getting rid of the additional `ConditionGroups` param for priority function which is difficult to understand without looking through the codebase.

I'm also open for any suggestion regarding how we should solve this problem.

Reply:
I think I will plus 1 to @iamhopaul123. A bonus for what boss suggests is ⬆️  easier for customers to do CDK override in case they really do hit the 100 rules limits